packages/dev/core/src/Meshes/mesh.vertexData.functions.ts

```diff
@@ -0,0 +1,75 @@
+import type { IndicesArray } from "core/types";
+
+/**
+ * Sort (in place) the index array so that faces with common indices are close
+ * @param indices the array of indices to sort
+ */
+export function OptimizeIndices(indices: IndicesArray) {
+    const faces: Array<Array<number>> = [];
+    const faceCount = indices.length / 3;
+
+    // Step 1: Break the indices array into faces
+    for (let i = 0; i < faceCount; i++) {
+        faces.push([indices[i * 3], indices[i * 3 + 1], indices[i * 3 + 2]]);
+    }
+
+    // Step 2: Build a graph connecting faces sharing a vertex
+    const vertexToFaceMap = new Map<number, number[]>();
+    faces.forEach((face, faceIndex) => {
+        face.forEach((vertex) => {
+            if (!vertexToFaceMap.has(vertex)) {
+                vertexToFaceMap.set(vertex, []);
+            }
+            vertexToFaceMap.get(vertex)!.push(faceIndex);

```

Comment:
Change to have one map lookup instead of two?
```suggestion
            let face = vertexToFaceMap.get(vertex);
            if (!face) {
                vertexToFaceMap.set(vertex, face = []);
            }
            face.push(faceIndex);
```